git.go

```diff
@@ -26,13 +32,130 @@ func (scm *GitScm) contributeOtelAttributes() []attribute.KeyValue {
 	if err != nil {
 		return gitAttributes
 	}
-	gitAttributes = append(gitAttributes, attribute.Key(ScmRepository).String(origin.Config().URLs[0]))
+	gitAttributes = append(gitAttributes, attribute.Key(ScmRepository).StringSlice(origin.Config().URLs))
 
 	branch, err := repository.Head()
 	if err != nil {
 		return gitAttributes
 	}
 	gitAttributes = append(gitAttributes, attribute.Key(ScmBranch).String(branch.Name().String()))
 
+	contributions := []func(*git.Repository) ([]attribute.KeyValue, error){
+		contributeCommitters,
+	}
+
+	for _, contribution := range contributions {
+		contributtedAttributes, err := contribution(repository)
+		if err != nil {
+			fmt.Printf(">> not contributing attributes: %v", err)
+			continue
+		}
+
+		gitAttributes = append(gitAttributes, contributtedAttributes...)
+	}
+
 	return gitAttributes
 }
+
+// contributeCommitters this method calculates the commits between current branch (HEAD) and a target branch.
+// - The target branch has to be set as the TARGET_BRANCH environment variable
```

Comment:
Is this env variable similar with other CI? Just wondering if TARGET_BRANCH differs in GitLab?

Reply:
They use `CI_MERGE_REQUEST_TARGET_BRANCH_NAME` I used `TARGET_BRANCH` so that the implementor should set that variable in the environment, but you are right, if we want to be CI agnostic, we should take care of predefined vars

Reply:
Added support for detecting the CI: Github Vs Gitlab